### ISSUE TITLE
Enable ingestion of compacted Debezium data

### ISSUE THREAD

**Issue by umanwizard** [MaterializeInc/materialize#3078]
One idea for how to do this is to keep track of primary keys until the initial ingest is finished, and then drop them.

**Comment by ruchirK** [comment]
is this effectively done by @quodlibetor 's work on debezium upsert in #5709

**Comment by umanwizard** [comment]
I disagree. IMO we should leave this open until it is doable without permanent memory overhead (which I think is eminently possible)

**Comment by quodlibetor** [comment]
How would we do it without permanent memory overhead? Are you talking about dropping the tracking state once we catch up to present?
